sdk/core/core-lro/src/azureAsyncOperationStrategy.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  LROStrategy,
+  BaseResult,
+  LROOperationStep,
+  LROResponseInfo,
+  FinalStateVia
+} from "./models";
+import { OperationSpec, OperationArguments } from "@azure/core-http";
+import { terminalStates } from "./constants";
+import { SendOperationFn } from ".";
+
+export function createAzureAsyncOperationStrategy<TResult extends BaseResult>(
+  initialOperation: LROOperationStep<TResult>,
+  sendOperationFn: SendOperationFn<TResult>,
+  finalStateVia?: FinalStateVia
+): LROStrategy<TResult> {
+  const lroData = initialOperation.result._lroData;
+  if (!lroData) {
+    throw new Error("Expected lroData to be defined for Azure-AsyncOperation strategy");
+  }
+
+  let currentOperation = initialOperation;
+  let lastKnownPollingUrl = lroData.azureAsyncOperation || lroData.operationLocation;
+
+  return {
+    isTerminal: () => {
```

Comment:
LRO's usually are "done" or not, is "Terminal" something else?

Reply:
`isTerminal` checks a few conditions to determine the path to follow, each strategy have different conditions to decide if the current state is terminal. To give better context, I'm attaching the flow-charts used as blueprints for these LRO implementations:

**General Flow:**
![image](https://user-images.githubusercontent.com/21109913/97921426-c878b180-1d0f-11eb-888a-b6b0b86733d6.png)

**Detailed Scenarios**
Strategies map the different paths from the original operation

![image](https://user-images.githubusercontent.com/21109913/97921526-ecd48e00-1d0f-11eb-8ec4-31217f4c4f1f.png)

Reply:
Let's schedule a meeting and talk this through!

Reply:
Makes sense I'll schedule some time this week. I'll leave the LRO section of Microsoft REST API guidelines here for reference: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations